File: email.js
document.addEventListener("DOMContentLoaded", function () {
    const form = document.getElementById("contactForm");
    const result = document.getElementById("result");

    if (!form) {
        console.error("contactForm not found");
        return;
    }

    form.addEventListener("submit", async function (e) {
        e.preventDefault();

        const data = {
            name: document.getElementById("name").value.trim(),
            email: document.getElementById("email").value.trim(),
            subject: document.getElementById("subject").value.trim(),
            description: document.getElementById("message").value.trim()
        };

        try {
            const response = await fetch(
                "https://xxxxxxxx.execute-api.ca-central-1.amazonaws.com/xxxxxxxx", /*Your api key*/
                {
                    method: "POST",
                    headers: {
                        "Content-Type": "application/json"
                    },
                    body: JSON.stringify(data)
                }
            );

            const responseText = await response.text();
            console.log("HTTP status:", response.status);
            console.log("Response body:", responseText);

            if (response.ok) {
                alert("Your email has been sent to Ivan Chiu.");
                form.reset();
                if (result) {
                    result.innerText = "Message sent successfully.";
                }
            } else {
                alert("Failed to send message. Please try again later.");
                if (result) {
                    result.innerText = "Failed to send message. " + responseText;
                }
            }
        } catch (error) {
            console.error("Error sending message:", error);
            alert("Network error. Check browser console.");
            if (result) {
                result.innerText = "Network error. Check browser console.";
            }
        }
    });
});

File: email.py
import json
import boto3

ses = boto3.client("ses", region_name="ca-central-1")  # use your SES region, e.g."us-east-1"

SENDER = "your-verified-sender@example.com"   # must be verified in SES
RECEIVER = "ylchiu1303@gmail.com"              # where you want to receive the message

def lambda_handler(event, context):
    try:
        body = json.loads(event.get("body", "{}"))

        user_email = body.get("email", "").strip()
        subject = body.get("subject", "").strip()
        message = body.get("message", "").strip()

        if not user_email or not subject or not message:
            return {
                "statusCode": 400,
                "headers": {
                    "Access-Control-Allow-Origin": "*",
                    "Access-Control-Allow-Headers": "Content-Type",
                    "Access-Control-Allow-Methods": "OPTIONS,POST"
                },
                "body": json.dumps({"error": "email, subject, and message are required"})
            }

        email_body = f"""
New message from website contact form

From user: {user_email}
Subject: {subject}

Message:
{message}
"""

        response = ses.send_email(
            Source=SENDER,
            Destination={"ToAddresses": [RECEIVER]},
            Message={
                "Subject": {"Data": f"Website Contact: {subject}"},
                "Body": {
                    "Text": {"Data": email_body}
                }
            },
            ReplyToAddresses=[user_email]
        )

        return {
            "statusCode": 200,
            "headers": {
                "Access-Control-Allow-Origin": "*",
                "Access-Control-Allow-Headers": "Content-Type",
                "Access-Control-Allow-Methods": "OPTIONS,POST"
            },
            "body": json.dumps({
                "message": "Email sent successfully",
                "messageId": response["MessageId"]
            })
        }

    except Exception as e:
        return {
            "statusCode": 500,
            "headers": {
                "Access-Control-Allow-Origin": "*",
                "Access-Control-Allow-Headers": "Content-Type",
                "Access-Control-Allow-Methods": "OPTIONS,POST"
            },
            "body": json.dumps({"error": str(e)})
        }


        
